sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -163,7 +163,12 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
         s"-- Number of queries: ${outputs.size}\n\n\n" +
         outputs.zipWithIndex.map{case (qr, i) => qr.toString(i)}.mkString("\n\n\n") + "\n"
       }
-      stringToFile(new File(testCase.resultFile), goldenOutput)
+      val resultFile = new File(testCase.resultFile);
+      val parent = resultFile.getParentFile();
+      if (!parent.exists()) {
+        assert(parent.mkdirs(), "Could not create directory: " + parent)
+      }
+      stringToFile(resultFile, goldenOutput)
```

Comment:
This addition is ported from the code from [PR-16467](https://github.com/apache/spark/pull/16467) of [SPARK-19017](https://issues.apache.org/jira/browse/SPARK-19017) reviewed by @hvanhovell.

This change is required after the introduction of test files in sub-directories by [SPARK-18871](https://issues.apache.org/jira/browse/SPARK-18871).